File: calculator/src/main/java/com/turlir/calculator/converter/Member.java
package com.turlir.calculator.converter;

import com.turlir.calculator.interpreter.NotationInterpreter;
import com.turlir.calculator.member.Operator;
import com.turlir.calculator.member.Value;

/**
 * Токен - участник математического выражения <br>
 * @see Value
 * @see Operator
 */
public interface Member {

    /**
     *
     * @return является ли токен операндом (числом, {@link Operator})
     */
    boolean operand();

    /**
     * Обработать токен согласно нотации
     * @param interpreter интерпретатор нотации для обработки части математического выражения
     */
    void process(NotationInterpreter interpreter);

    /**
     *
     * @return экземпляр интерфейса для представления токена в {@link Printer}
     */
    Visual view();

}

File: calculator/src/main/java/com/turlir/calculator/converter/Member.kt
package com.turlir.calculator.converter

import com.turlir.calculator.interpreter.NotationInterpreter
import com.turlir.calculator.member.Operator
import com.turlir.calculator.member.Value

/**
 * Токен - участник математического выражения <br></br>
 * @see Value
 * @see Operator
 */
interface Member {

    /**
     *
     * @return является ли токен операндом (числом, [Operator])
     */
    fun operand(): Boolean

    /**
     * Обработать токен согласно нотации
     * @param interpreter интерпретатор нотации для обработки части математического выражения
     */
    fun process(interpreter: NotationInterpreter)

    /**
     *
     * @return экземпляр интерфейса для представления токена в [Printer]
     */
    fun view(): Visual

}
